### PR TITLE
feat: add pubsublite.ordering.mode to kafka connector

### PR DESCRIPTION
**Pull request by dpcollins-google**
This is useful for migration cases using the kafka wire protocol.

Also clean up dependency management and upgrade pubsublite and other versions.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -15,54 +15,73 @@
   <parent>
     <groupId>com.google.cloud</groupId>
     <artifactId>google-cloud-shared-config</artifactId>
-    <version>1.5.4</version>
```
Comment (samarthsingal) [inline pom.xml:18]: Why are we regressing the version here?

File: pom.xml [hunk 2]
```diff
@@ -15,54 +15,73 @@
...
+    <version>1.5.5</version>
   </parent>
 
   <properties>
     <maven.compiler.source>1.8</maven.compiler.source>
     <maven.compiler.target>1.8</maven.compiler.target>
 
-    <kafka.version>3.3.1</kafka.version>
```
Comment (samarthsingal) [inline pom.xml:25]: Why are we getting rid of version properties? This is the recommended POM structure for GA libraries I was told.

Comment (samarthsingal) [reply pom.xml:25]: In general, can you separate out POM changes you are making for this change and the ones that are purely stylistic? Or are all these changes needed as-is?

File: pom.xml (hunk 1 above)
Comment (dpcollins-google) [reply pom.xml:18]: We're increasing the version.

File: src/main/java/com/google/pubsublite/kafka/sink/PublisherFactoryImpl.java [hunk 3]
```diff
@@ -82,8 +80,7 @@ public Publisher<MessageMetadata> newPublisher(Partition partition) throws ApiEx
                     responseStream -> {
                       ApiCallContext context =
                           getCallContext(
-                              PubsubContext.of(Framework.of(PUBSUBLITE_KAFKA_SINK_CONNECTOR_NAME)),
```
[the remark this answers is not in the document]
Comment (dpcollins-google) [reply src/main/java/com/google/pubsublite/kafka/sink/PublisherFactoryImpl.java:85]: This is interpreted by the backend and converted to an enum. I need to add it to the enum, and so I'm making it have the same structure as the rest of them.

File: pom.xml (hunk 2 above)
Comment (dpcollins-google) [reply pom.xml:25]: This also isn't the recommended way to define dependency versions for shared dependencies- I'm converting this to follow the best practice.

File: src/main/java/com/google/pubsublite/kafka/sink/OrderingMode.java [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.pubsublite.kafka.sink;
+
+public enum OrderingMode {
+  DEFAULT,
```
[the remark this answers is not in the document]
Comment (dpcollins-google) [reply src/main/java/com/google/pubsublite/kafka/sink/OrderingMode.java:20]: Done.

File: pom.xml (hunk 2 above)
Comment (samarthsingal) [reply pom.xml:25]: Discussed offline, this now uses BOM structure - https://cloud.google.com/java/docs/bom.

**Review by samarthsingal — APPROVED** [review]

**Pull request merged**